lib/querytables.js

```diff
@@ -245,61 +242,59 @@ async function setUpdateAtToNonRegularTables (pg, rows) {
  * - `_cdb_dbname_of_foreign_table`
  * - `CDB_Get_Foreign_Updated_At` via getForeignTableUpdatedAt
  */
-function getTablesMetadata (pg, tableArray, callback) {
-    /* Note: We order by **reloid** because that's the implicit behaviour of CDB_QueryTables_Updated_At
-     * Eventhough `CDB_QueryTablesText` orders alphabetically (by our id_name), the unnest call breaks
-     * that ordering and somehow (PG internals) the subsequent calls end up ordering the table names
-     * by their `::regclass::oid` */
-
-    let metadataQuery = `WITH cdb_table_names AS (
-        SELECT format('%s.%s', quote_ident('${encodePGName(tableArray[0].schema_name)}'),
-                               quote_ident('${encodePGName(tableArray[0].table_name)}')) as id_name
-    `;
-
-    for (let i = 1; i < tableArray.length; i++) {
-        metadataQuery += `UNION ALL
-        SELECT format('%s.%s', quote_ident('${encodePGName(tableArray[i].schema_name)}'),
-                               quote_ident('${encodePGName(tableArray[i].table_name)}')) as id_name
+async function getTablesMetadata (pg, tableArray) {
+    try {
```

Comment:
Can you move the try-catch to catch just the error that you need, i.e. the queries instead of the whole function?